src/Service.Tests/SqlTests/RestApiTests/Put/PutApiTestBase.cs

```diff
@@ -156,27 +156,50 @@ public virtual async Task PutOne_Update_IfMatchHeaders_Test()
         }
 
         /// <summary>
-        /// Test to validate that PUT update on a row accessible to the user after applying database policy
-        /// filters executes successfully.
+        /// Test to validate successful execution of PUT operation which satisfies the database policy for the operation (insert/update) it resolves into.
         /// </summary>
-        /// <returns></returns>
         [TestMethod]
-        public virtual async Task PutOneUpdateAccessibleRowWithDatabasePolicy()
+        public virtual async Task PutOneWithDatabasePolicy()
         {
+            // PUT operation resolves to update because we have a record present for given PK.
+            // Since the database policy for update operation ("@item.pieceid ne 1") is satisfied by the operation, it executes successfully.
             string requestBody = @"
             {
-                ""name"": ""New Publisher""
+                ""piecesAvailable"": 4,
+                ""piecesRequired"": 5,
```

Comment:
Can you please change the update that would happen here to reflect PUT's "replace" like behavior?

Dont provide one of the non-required columns and expect it becomes a NULL on the backend...